gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,322 @@
+                                  startedAt,
+                                  refName,
+                                  contentId,
+                                  contentValues,
+                                  objectMapper))));
+    }
+    return rows.iterator();
+  }
+
+  private static Row getMarkerRow(String runID, Timestamp startedAt) {
+    return RowFactory.create(
+        IdentifiedResultsRepo.TYPE_GC_MARKER, startedAt, runID, null, null, null);
+  }
+
+  private static Row getContentRow(
+      String runID,
+      Timestamp startedAt,
+      String refName,
+      String contentId,
+      ContentValues contentValues,
+      ObjectMapper objectMapper) {
+    return RowFactory.create(
+        IdentifiedResultsRepo.TYPE_CONTENT,
+        startedAt,
+        runID,
+        contentId,
+        contentValues.getExpiredContents().stream()
+            .map(
+                content -> {
+                  try {
+                    // serialize to json and store the contents
+                    return objectMapper.writeValueAsString(content);
+                  } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                  }
+                })
+            .toArray(String[]::new),
+        refName);
+  }
+
+  private static StructType checkAndCreateTable(
+      SparkSession sparkSession,
+      String catalogName,
+      TableIdentifier tableIdentifier,
+      String catalogAndTable,
+      String gcRefName) {
+    // get Nessie catalog
+    Catalog nessieCatalog =
+        CatalogUtil.loadCatalog(
+            NessieCatalog.class.getName(),
+            catalogName,
+            catalogConfWithRef(sparkSession, catalogName, gcRefName),
+            sparkSession.sparkContext().hadoopConfiguration());
+    // create table if not exists
+    sparkSession.sql(String.format(SCHEMA_DDL, withRefName(catalogAndTable, gcRefName)));
+    // get schema
+    return SparkSchemaUtil.convert(nessieCatalog.loadTable(tableIdentifier).schema());
+  }
+
+  private static void addRowToIdentifiedResults(
+      IdentifiedResult identifiedResult, ObjectMapper objectMapper, Row row) {
+    List<Object> contentsInRow = row.getList(1);
+    List<Content> contents =
+        contentsInRow.stream()
+            .map(val -> (String) val)
+            .map(content -> parseContentValues(content, objectMapper))
+            .collect(Collectors.toList());
+    contents.forEach(content -> identifiedResult.addContent(row.getString(2), content));
+  }
+
+  private static Content parseContentValues(String content, ObjectMapper objectMapper) {
+    try {
+      return objectMapper.readValue(content, Content.class);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static String withRefName(String catalogAndTable, String refName) {
+    int tableNameIndex = catalogAndTable.lastIndexOf(".") + 1;
+    return catalogAndTable.substring(0, tableNameIndex)
```

Comment:
why not set the context via teh spark session or store the underlying as member variables? I don't know why we are doing string manipultion here. And I don't like explicitly relying on the `@` syntax

Reply:
Don't want to affect other concurrent operation on this spark context by setting a `ref`, currently directly setting for each table ops using the `@` syntax.

I will make it a member variable and use a `TableReference` object, that internally takes care about string manipulation.